pandas/tests/dtypes/test_dtypes.py

```diff
@@ -451,26 +451,41 @@ def test_hash_vs_equality(self):
         assert hash(dtype2) == hash(dtype2)
         assert hash(dtype2) == hash(dtype3)
 
-    def test_construction(self):
-        with pytest.raises(ValueError):
-            IntervalDtype('xx')
-
-        for s in ['interval[int64]', 'Interval[int64]', 'int64']:
-            i = IntervalDtype(s)
-            assert i.subtype == np.dtype('int64')
-            assert is_interval_dtype(i)
-
-    def test_construction_generic(self):
-        # generic
-        i = IntervalDtype('interval')
-        assert i.subtype == ''
+    @pytest.mark.parametrize('subtype', [
+        'interval[int64]', 'Interval[int64]', 'int64', np.dtype('int64')])
```

Comment:
hmm, I though we *only* accept ``interval[...]`` and not ``Interval[...]``?

Reply:
The existing behavior allows for `Interval[...]`.  Really just took those subtypes out of a `for` loop and used `parametrize` instead in an effort to clean up the test.  The existing regex match string includes "I", and there is existing code checks against `.title()`, so it looks like this was intentional:

https://github.com/pandas-dev/pandas/blob/84335621ad0a0d83302a80b6911d3985c00b5cee/pandas/core/dtypes/dtypes.py#L635

https://github.com/pandas-dev/pandas/blob/84335621ad0a0d83302a80b6911d3985c00b5cee/pandas/core/dtypes/dtypes.py#L708-L709

I think this was copied from `PeriodDtype`, because I see nearly identical code there.  Doesn't look like it'd be too hard to remove this to enforce lowercase, if so desired.